wallet/internal/db/itest/wallet_store_test.go

```
@@ -0,0 +1,277 @@
+//go:build itest
+
+package itest
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcwallet/wallet/internal/db"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCreateWallet verifies that CreateWallet correctly creates a wallet
+// and returns its information.
+func TestCreateWallet(t *testing.T) {
+	t.Parallel()
+
+	store, _ := NewTestStore(t)
+	params := CreateWalletParamsFixture("test-wallet")
+
+	info, err := store.CreateWallet(t.Context(), params)
+	require.NoError(t, err)
+	require.NotNil(t, info)
+
+	require.Equal(t, info.ID, uint32(1), "first wallet ID should be 1")
+	require.Equal(t, params.Name, info.Name)
+	require.Equal(t, params.IsImported, info.IsImported)
+	require.Equal(t, params.ManagerVersion, info.ManagerVersion)
+	require.Equal(t, params.IsWatchOnly, info.IsWatchOnly)
+
+	require.Nil(t, info.SyncedTo)
+	require.Equal(t, uint32(0), info.BirthdayBlock.Height)
+}
+
+// TestCreateWallet_DuplicateName verifies that creating a wallet with a
+// duplicate name fails with an appropriate error.
+func TestCreateWallet_DuplicateName(t *testing.T) {
+	t.Parallel()
+
+	store, _ := NewTestStore(t)
+	params := CreateWalletParamsFixture("duplicate-wallet")
+
+	_, err := store.CreateWallet(t.Context(), params)
+	require.NoError(t, err)
+
+	// Attempt to create second wallet with same name.
+	_, err = store.CreateWallet(t.Context(), params)
+	require.Error(t, err, "expected error creating duplicate wallet")
```

We should perform more rigorous check here, so check the error type via `require.ErrorIs`?

---

It is a bit tricky because SQLite returns `constraint failed: UNIQUE constraint failed: wallets.name (2067)` while Postgres returns `ERROR: duplicate key value violates unique constraint "uidx_wallets_name" (SQLSTATE 23505)`. I added a check that asserts the error contains the common parts of these messages, please see if that feels reliable enough. If not, we will probably need to handle and map these errors in a more structured way.

---

yeah ideally we don't want to return the raw error from sql here, and we should wrap it so the upper layer understands how it should be handled - but that can be done in future improvement PRs.